src/libraries/Microsoft.Extensions.Logging.Abstractions/gen/LoggerMessageGenerator.Parser.cs

```diff
@@ -450,13 +455,13 @@ public IReadOnlyList<LoggerClass> GetLogClasses(IEnumerable<ClassDeclarationSynt
                     switch (a.NameEquals.Name.ToString())
                     {
                         case "EventId":
-                            eventId = (int)sm.GetConstantValue(a.Expression, _cancellationToken).Value!;
+                            eventId = (int?)sm.GetConstantValue(a.Expression, _cancellationToken).Value ?? -1;
```

Comment:
Related to issue https://github.com/dotnet/runtime/issues/52223
When event id is not set, it will default to -1.